ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1182,13 +1182,23 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 	// TODO (kevin85421): ObservedGeneration should be used to determine whether to update this CR or not.
 	newInstance.Status.ObservedGeneration = newInstance.ObjectMeta.Generation
 
-	runtimePods := corev1.PodList{}
-	filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: newInstance.Name}
-	if err := r.List(ctx, &runtimePods, client.InNamespace(newInstance.Namespace), filterLabels); err != nil {
+	headPods := corev1.PodList{}
```

Comment:
There are several advantages to separately getting head Pods and worker Pods.

1. When calculating readyWorkerReplicas and AvailableReplicas, if we initially get all Ray Pods, we still need to distinguish which are worker Pods and may use a filter label again.
2. The same thing happens when we need to determine whether the head pod is ready or not.